Add 'OUT_DIR' to quick start and tutorial (#875)
lalrpop/lalrpop: doc/src/tutorial/001_adding_lalrpop.md

```

The function `process_root` processes your `src` directory, converting
all `lalrpop` files into `rs` files, and saving them to `OUT_DIR`. It is smart
enough to check timestamps and do nothing if the `rs` file is newer than the
`lalrpop` file, and to mark the generated `rs` file as read-only. It returns an
`io::Result<()>`, so the `unwrap()` call just asserts that no
file-system errors occurred.

You can include the generated code somewhere in your Rust code. For example, if

> the source grammar were to be called `grammar.lalrpop`, you could include the
> generated `grammar.rs` file like so:

Reviewer: ```suggestion
the source grammar is in `grammar.lalrpop`, you can include the
generated `grammar.rs` file like so:
```

```rust

> include!(concat!(env!("OUT_DIR"), "/grammar.rs"));

Reviewer: What is the purpose of including the generated gammar like this? The wording sounds like this is the normal way of using the generated grammar, while one should probably prefer `lalrpop_mod!` instead.

Reviewer 2: Could also link to the https://lalrpop.github.io/lalrpop/generate_in_source.html section as a way to get similar behavior I think.

Author: This is the very first time I've hearing of `lalrpop_mod!`. The documentation search shows it appears in a few code samples without explicitly mentioning what it does: https://lalrpop.github.io/lalrpop/index.html?search=Lalrpop_mod

(Only one example references it in text, saying 'it's not useful' when generating to the source directory.)

Author: (So yeah, this should probably appear more often in the docs. I'll update my PR when I have the chance.)

Reviewer: Oh, I thought `lalrpop_mod` was part of the documentation in a clear way...thanks @galexite, mentioning `lalrpop_mod` is a most wanted addition then. I just assumed `lalrpop_mod` was known and that you wanted to achieve something different, but I couldn't really figure out what.

```

The function `process_root` processes your `src` directory, converting
all `lalrpop` files into `rs` files, and saving them to `OUT_DIR`. It is smart
enough to check timestamps and do nothing if the `rs` file is newer than the
`lalrpop` file, and to mark the generated `rs` file as read-only. It returns an
`io::Result<()>`, so the `unwrap()` call just asserts that no
file-system errors occurred.

> You can include the generated code somewhere in your Rust code. For example, if
> the source grammar is in `grammar.lalrpop`, you can add the generated
> `grammar.rs` file as a module using the [`lalrpop_mod!`][lalrpop_mod] macro:

Reviewer: ```suggestion
The [`lalrpop_mod!`][lalrpop_mod] macro generates a wrapper module in your
crate so that you can use the generated parser from your code. For example,
if the source grammar is located in `grammar.lalrpop`, adding the following line
to `lib.rs` will create a corresponding `grammar` submodule (note that you can
also add this line to a `foo.rs` module definition instead, which will then
create a submodule `foo::grammar`):
```
